helios-services/src/main/java/com/spotify/helios/master/resources/HostsResource.java

```diff
@@ -147,13 +148,21 @@ public HostDeregisterResponse delete(@PathParam("id") final String host) {
   @Produces(APPLICATION_JSON)
   @Timed
   @ExceptionMetered
-  public Optional<HostStatus> hostStatus(@PathParam("id") final String host) {
-    return Optional.fromNullable(model.getHostStatus(host));
+  public Optional<HostStatus> hostStatus(@PathParam("id") final String host,
+                                         @QueryParam("status") final String statusFilter) {
```

Comment:
It should work. Here's a test: https://github.com/spotify/helios/pull/587/files#diff-dc34e9cadd23a8d110d25e12c41041e8R81